controllers/project_controller_test.go

```diff
@@ -115,6 +122,9 @@ var _ = Describe("Project Controller", func() {
 		Describe("Create new Jira servie desk project resource", func() {
 			Context("with invalid fields", func() {
 				It("should not create a new project", func() {
+					key := cUtil.RandSeqString(10)
+					projectInvalidInput.Spec.Key = strings.ToUpper(key)
```

Comment:
Key can not be of 10 digits. I think you replaced the value for key with name.

Reply:
I changed key size to be 9 and projectInvalidInput.Spec.Name += key[:3] this line just appends a random 3 char long string to existing name.